modules/features2d/src/sift.simd.hpp

```diff
@@ -566,13 +566,13 @@ void calcSIFTDescriptor(
     float bins_per_rad = n / 360.f;
     float exp_scale = -1.f/(d * d * 0.5f);
     float hist_width = SIFT_DESCR_SCL_FCTR * scl;
-    int radius = cvRound(hist_width * 1.4142135623730951f * (d + 1) * 0.5f);
+    int radius = cvRound(hist_width * 1.4142135623730951f * (d + 1) * 0.5f + 0.5f);
```

Comment:
`static_cast<int>(hist_width * 1.4142135623730951f * (d + 1) * 0.5f + 0.5f)` (from `opensift`)

and

`cvRound(hist_width * 1.4142135623730951f * (d + 1) * 0.5f)`

already provides the similar result (`cvRound()` has addition of `+0.5` internally for values `> 0`)

So this change (extra `+0.5`) introduces more difference than without of this patch.

---

Currently there is no goal to provide bit-exact result of SIFT, because this code is based on floating-point computations (they are not 100% equal between platforms or compiler optimizations).

Reply:
Got it. Thank you! Will close.